### PR TITLE
Tests/refactor

### PR DESCRIPTION
**Pull request by pmast**

### CODE REVIEW

File: src/geocoder.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ export default class Geocoder {
    * Constructs a geocoder.
    * @param  {Object} options Geocoder options.
    */
-  constructor(options = {}) {
+  constructor(options = {}, geoCoder = GoogleGeocoder, GeoCache = Cache) {
```
Comment (ro-ka) [inline src/geocoder.js]: Better name it `geocoder`, as it is not camelCased in other occurences neither.

File: src/index.js [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
+ *                           privateKey: null}
+ * @param {Object} Geocoder A geocoder.
+ * @param {Object} GeocodeStream A GeocodeStream.
  */
 export default class GeoBatch {
-  constructor({cacheFile, clientId, privateKey} =
-    {cacheFile: 'geocache.db', clientId: null, privateKey: null}) {
+  constructor(
+    {cacheFile = 'geocache.db', clientId = null, privateKey = null}
+      = {cacheFile: 'geocache.db', clientId: null, privateKey: null},
```
Comment (ro-ka) [inline src/index.js:20]: Duplicate line?

File: src/index.js [hunk 3]
```diff
@@ -1,18 +1,27 @@
...
+    Geocoder = StandardGeocoder,
+    GeocodeStream = StandardGeocodeStream) {
```
Comment (ro-ka) [inline src/index.js]: Break `) {` to a new line so it is easier to see the param block and the content block.

File: src/index.js [hunk 4]
```diff
@@ -22,15 +31,25 @@ export default class GeoBatch {
...
+   * @param  {Stream} stream An input stream
+   * @param  {Object} stats  An object with the stream stats, defaults to {}.
+   * @return {Stream}        A transformable stream.
+   */
+  geocodeStream(stream, stats = {}) {
+    const streamGeocoder = new this.GeocodeStream(this.geocoder, stats);
+    stream.pipe(streamGeocoder);
 
-    return geocodeStream;
+    return streamGeocoder;
```
Comment (ro-ka) [inline src/index.js]: Namings are a bit confusing: geocodeStream, streamGeocoder…

File: src/index.js (hunk 2 above)
Comment (pmast) [inline src/index.js:20]: No... Horrible construction. But:

The first is destructuring with default values for if an object is given.
The second line is the default if nothing is given.

File: src/geocoder.js [hunk 5]
```diff
@@ -33,11 +33,8 @@ export default class Geocoder {
     this.lastGeocode = new Date();
     this.currentRequests = 0;
 
-    this.cache = new Cache(options.cacheFile);
-    this.googlemaps = googlemaps;
-
-    this.googlemaps.config('google-client-id', options.clientId);
-    this.googlemaps.config('google-private-key', options.privateKey);
+    this.cache = new GeoCache(options.cacheFile);
+    this.googlemaps = geoCoder.init(options);
```
Comment (ro-ka) [inline src/geocoder.js]: Either the var is indicating that it is a Google lib or not. Naming is inconsistent: googleMaps, geoCoder, googlemaps for the same thing.

File: src/lib/google-geocoder.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import googleMaps from 'googlemaps';
+
+export default class GoogleGeocoder {
+  static init({clientId, privateKey} = {clientId: null, privateKey: null}) {
```
Comment (ro-ka) [inline src/lib/google-geocoder.js:10]: Add some comment. :)

**Comment by ro-ka** [comment]
:checkered_flag:

**Comment by pmast** [comment]
:grapes:

**Comment by ro-ka** [comment]
You are allowed to release. :)

**Pull request merged**